src/System.Runtime.Extensions/src/System/IO/StreamReader.cs

```diff
@@ -450,6 +461,24 @@ public override int ReadBlock(char[] buffer, int index, int count)
             return base.ReadBlock(buffer, index, count);
         }
 
+        public override int ReadBlock(Span<char> buffer)
+        {
+            if (GetType() != typeof(StreamReader))
+            {
+                // Defer to Read(char[], ...) if a derived type may have previously overridden it.
+                return base.ReadBlock(buffer);
+            }
+
+            int i, n = 0;
+            do
+            {
+                i = ReadSpan(buffer);
```

Comment:
Should ReadBlock take in a ReadOnlySpan instead of a Span?

Also, how about re-writing this as follows instead?
```C#
{
i = ReadSpan(buffer.Slice(n));
n += i;
}
```

Reply:
No, it's reading into the span; the span needs to be writable.

Reply:
> Also, how about re-writing this as follows instead?

We could... is there a benefit to that?

Reply:
It looks like the assembly is smaller.

![image](https://user-images.githubusercontent.com/6527137/31789107-db1c5c7a-b4c5-11e7-8608-d3e426fdf592.png)

Reply:
Ok, I'll change it.